.github/CONTRIBUTING.md

```diff
@@ -159,6 +160,47 @@ This can happen for several reasons:
 To exclude a task from validation, open the corresponding Jupyter notebook and choose ```View -> Cell Toolbar -> Tags``` to see and edit the tags for each cell. Add the tag that is the most fitting description of the failure cause to the cell. 
 After you are done with editing the notebook, choose ```View -> Cell Toolbar -> None``` to turn off tags editing view for the subsequent users of this notebook. Finally, save the notebook.
 
+#### Validate changes to `%kata` magic and `%check-kata` magic on local machine
+1. Add logging statements to the [`KataMagic`](./utilities/Microsoft.Quantum.Katas/KataMagic.cs) and [`CheckKataMagic`](./utilities/Microsoft.Quantum.Katas/CheckKataMagic.cs).
+   ```
+   Logger.LogDebug($"Value : {val}");
+   ```
+2. Generate a custom nuget package
+   1. Build the Microsoft.Quantum.Katas project to produce a NuGet package to get a version other than 1.0.0 by adding the following code in [this file](./utilities/Microsoft.Quantum.Katas/Microsoft.Quantum.Katas.csproj) by adding the following code
+   ```
+   <RootNameSpace>Microsoft.Quantum.Katas</RootNameSpace>
+   <Version>version-number</Version> ).
+   <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+   ```
+   2. Copy the generated .nupkg file to the folder in which your project resides (like PhaseEstimation kata) and remove the following line from your project's project file.
+   ```
+   <PackageReference Include="Microsoft.Quantum.Katas" Version="0.16.2104138035" />
+   ```
+   3. Set up NuGet.config file so that it allows the project to discover a NuGet package  in the current folder in addition to   the standard sources. Here's an example file    (should be placed in project folder as well):
+   ```
+   <?xml version="1.0" encoding="utf-8"?>
+   <configuration>
+     <packageSources>
+       <add key="Local Folder" value="." />
+     </packageSources>
+   </configuration>
+   ```
+3. To observe your changes in console window where Jupyter Notebook was launched.
+   1. Set the environment variable `IQSHARP_LOG_LEVEL='Debug'`.
+   > Note : The environment variable is case-sensitive.
+   2. Navigate to the folder in which your project resides and do the following :
+   ```
+   $ start jupyter notebook
+   ```
+   > This would launch Jupyter Notebooks server in a new window which would be
+   > helpful in debugging the magics since logging is a bit noisy.
+
+4. Changes to be done in jupyter notebook 
+   1. Use `%package` magic to load the newly built nuget package `Microsoft.Quantum.Katas` by adding the following code
+   ```
+   %package Microsoft.Quantum.Katas::<package-version>
```

Comment:
Do we need this line, or would it work if we just change the version in the `<PackageReference Include="Microsoft.Quantum.Katas" Version="0.16.2104138035" />` dependency in the .csproj file? Trying to figure out how to minimize the number of steps

Reply:
I initially tried that approach only but the IQ# kernel for no particular reason loaded the katas package of the same version as the IQ# kernel. I think this might be due to the fact that IQ# kernel would attempt to load dependencies of the same version as the kernel if <IQSharpLoadAutomatically> property is set to true.